### ISSUE TITLE
For Rails 6.1.0.alpha Devise version is 1.5.4 and undefined method `alias_method_chain’

### ISSUE THREAD

**Issue by efojs** [rails/rails#36673]
### Steps to reproduce
Install master branch: **rails (6.1.0.alpha)**, ruby 2.5.5

Create new app: 
`$ rails new APP_NAME --database=postgresql --skip-sprockets --skip-test --webpack=react`

Add Devise: 
```
# Gemfile
gem 'devise'
$ bundle
```

### Expected behavior — latest Devise version (4.6.2) or at least > 4.5.0
`$ gem list`
` devise (4.6.2)`

OR

`$ bundle show devise`
`/Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-4.6.2`

OR

Run `$ rails generate devise:install` should result in some success (I'm not sure which)
Or (at least) Rails console:
```
$ rails c
Running via Spring preloader in process 9880
Loading development environment (Rails 6.1.0.alpha)
2.5.5 :001 > 
```

OR 

Add version constraint:
```
# Gemfile
gem 'devise', '~> 4.6.2'

$ bundle
Bundle complete! 15 Gemfile dependencies, 73 gems now installed.
Use 'bundle info [gemname]' to see where a bundled gem is installed.
```

### Actual behavior — old Devise version (1.5.4) or crashes
`gem list`
`devise (1.5.4)`

OR
 
`$ bundle show devise`
`/Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4`

OR

Run `$ rails generate devise:install` or Rails console:
```
$ rails c
/Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4/lib/devise/rails/routes.rb:14:in `<class:RouteSet>’: undefined method `alias_method_chain’ for ActionDispatch::Routing::RouteSet:Class
Did you mean?  alias_method (NoMethodError)
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4/lib/devise/rails/routes.rb:2:in `<module:Routing>’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4/lib/devise/rails/routes.rb:1:in `<top (required)>’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:301:in `block in require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:268:in `load_dependency’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:301:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4/lib/devise/rails.rb:1:in `<top (required)>’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:301:in `block in require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:268:in `load_dependency’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/activesupport-6.1.0.alpha/lib/active_support/dependencies.rb:301:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/devise-1.5.4/lib/devise.rb:445:in `<top (required)>’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `block in require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/loaded_features_index.rb:92:in `register’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require_with_bootsnap_lfi’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/bootsnap-1.4.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler/runtime.rb:81:in `block (2 levels) in require’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler/runtime.rb:76:in `each’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler/runtime.rb:76:in `block in require’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler/runtime.rb:65:in `each’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler/runtime.rb:65:in `require’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/bundler.rb:114:in `require’
	from /Users/USERNAME/_dev/rails/APP_NAME/config/application.rb:20:in `<top (required)>’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:92:in `require’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:92:in `preload’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:157:in `serve’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:145:in `block in run’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:139:in `loop’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application.rb:139:in `run’
	from /Users/USERNAME/.rvm/gems/ruby-2.5.5@rails6.1_APP_NAME/gems/spring-2.1.0/lib/spring/application/boot.rb:19:in `<top (required)>’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:54:in `require’
	from /Users/USERNAME/.rvm/rubies/ruby-2.5.5/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:54:in `require’
	from -e:1:in `<main>’
```

OR

Add version constraint:
```
# Gemfile
gem 'devise', '~> 4.6.2'

$ bundle
Resolving dependencies...
Bundler could not find compatible versions for gem "railties":
  In snapshot (Gemfile.lock):
    railties (= 6.1.0.alpha)

  In Gemfile:
    devise (~> 4.6.2) was resolved to 4.6.2, which depends on
      railties (>= 4.1.0, < 6.0)

    rails (~> 6.1.0.alpha) was resolved to 6.1.0.alpha, which depends on
      railties (= 6.1.0.alpha)

    rspec-rails (~> 3.8) was resolved to 3.8.2, which depends on
      railties (>= 3.0)

    web-console (>= 3.3.0) was resolved to 4.0.0, which depends on
      railties (>= 6.0.0.a)

    webpacker (~> 4.0) was resolved to 4.0.7, which depends on
      railties (>= 4.2)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

```

### System configuration
**Rails version**:
rails (6.1.0.alpha)

**Ruby version**:
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-darwin15]

---
Edit: excuse me if it's on Devise side, IDK, is it?
Edit2: add `$ rails generate devise:install`

**Comment by efojs** [comment]
Also after `$ bundle` without version constraint get:
```
#######################################################

The bcrypt-ruby gem has changed its name to just bcrypt.  Instead of
installing `bcrypt-ruby`, you should install `bcrypt`.  Please update your
dependencies accordingly.

#######################################################

```

**Comment by wjessop** [comment]
I was able to repro this given the directions.

@efojs It looks like Rails 6.0 (no mention of 6.1 at all) support hasn't been released as a gem yet for devise, from the [changelog](https://github.com/plataformatec/devise/blob/master/CHANGELOG.md)

> ### Unreleased
> 
> * enhancements
>   * Support Rails 6.0

though when I try this experiment with Rails 6.0.0.rc1 it works as expected, isntalling devise 4.6.2. However, try adding this to your Gemfile:

`gem 'devise', git: 'https://github.com/plataformatec/devise.git'`

the running `bundle` in your project. For me `bundle show` gave me devise head:

```
$ bundle show devise
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
/Users/will/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/bundler/gems/devise-19045a483dd0
```

(19045a483dd0 is last commit to devise currently) and both `rails generate devise:install` and `rails c` work without error.

I think the issue here is the version specs in devise. If I pin the devise version to the latest in the gemfile:

`gem 'devise', '4.6.2'`

then `bundle update` I get this error:

```
Bundler could not find compatible versions for gem "activesupport":
  In Gemfile:
    jbuilder (~> 2.7) was resolved to 2.9.1, which depends on
      activesupport (>= 4.2.0)

    rails (~> 6.1.0.alpha) was resolved to 6.1.0.alpha, which depends on
      activesupport (= 6.1.0.alpha)

Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    devise (= 4.6.2) was resolved to 4.6.2, which depends on
      railties (>= 4.1.0, < 6.0)

    rails (~> 6.1.0.alpha) was resolved to 6.1.0.alpha, which depends on
      railties (= 6.1.0.alpha)

    web-console (>= 3.3.0) was resolved to 4.0.0, which depends on
      railties (>= 6.0.0.a)
```

The key here I think is:

```
    devise (= 4.6.2) was resolved to 4.6.2, which depends on
      railties (>= 4.1.0, < 6.0)
```

This looks to me like something that will resolve itself when devise release a new gem peraps, you might ask then about this issue.

**Comment by efojs** [comment]
Thank you for your time, @wjessop, cool! It's working as expected.


> from the changelog    

TIL a bit more where and how to check
